### DISCUSSION TITLE
Future of the project

### DISCUSSION THREAD

**Discussion by michaelb** [michaelb/sniprun#35] [General]
As the original project developer, this is how i envision sniprun's future:

The following is already a reality, and I'd like it to stay that way

- Simple in utilisation and understanding
- Allow simple contribution (addition of new languages)
- Documentation effort 
- Sniprun only runs the code it needs to run; no unecessary imports etc..

This would be nice to have:

- Passing arguments from vimscript to interpreters
- Optional dependencies (features?) (for in-development interpreters)
- Auto-install dependencies (but high risk to collide with the user's personnal config)
- Popularity (more contributors would be welcome)
- Test automation (WIP)
-̶ ̶l̶e̶t̶ ̶c̶o̶n̶f̶i̶g̶u̶r̶a̶t̶i̶o̶n̶=̶o̶p̶t̶i̶o̶n̶s̶ ̶i̶n̶ ̶v̶i̶m̶ done
- integrated parsing framework (to help implementation of higher support levels)

Seeing how there are few satisfying alternatives (comment if i am wrong), especially for compiled languages, and how quickly i could develop this by myself * , I'd like to see sniprun become a popular plugin. Maybe the all-language partial tester? ? I can dream alright.

**Comment by denstiny** [comment]
I think if sniprun can help me to test the practicability of a function, he will become the mainstream plug-in

**Comment by denstiny** [comment]
![image](https://user-images.githubusercontent.com/57088952/111284059-b96d1d00-867a-11eb-8e50-c807bc237e67.png)

**Reply by denstiny** [reply]
But can you tell me how to do

**Reply by michaelb** [reply]
That's in the plans and was envisaged from the start !!

I've done some conclusive tests (auto function fetching), but I need a stable treesitter interface (due for neovim 0.5) before I can really work on this with confidence.

At "File" support level, Sniprun would be run from only the `printf` line. It will then understand it needs the variables i and b, and thus search for the definition of i and b. Sniprun would also include (intelligently) everything it needs between (because side effects may occur like presently) the printf and the definition and then understand it needs the function `Swap`, and fetch the definition. 

Partial "File" support may be a bit impredictable / inconsistent if not done right, so I won't do a half-assed release.
Some more people are interested (with reason) to see this happen, and I'm quite hyped myself, though the technical challenge will be great - you won't see this tomorrow.

**Reply by denstiny** [reply]
I'm looking forward to, but I can't write a vim plug-in, or you really want to really help you

**Reply by michaelb** [reply]
Don't worry about it. Not many people (incl. me) are confident in writing a lua treesitter interface + communication with the Rust code. Programming is learning, give me enough time and i'll figure it out.
